webapp/src/pages/Settings.tsx

```diff
@@ -207,6 +217,17 @@ const Settings: React.FC = () => {
     Partial<AzimuthConfig>
   >({});
 
+  React.useEffect(() => {
+    if (defaultConfig && language !== undefined) {
+      setPartialConfig({
+        ...partialConfig,
+        syntax: defaultConfig.syntax,
+        similarity: defaultConfig.similarity,
+        behavioral_testing: defaultConfig.behavioral_testing,
+      });
+    }
+  }, [defaultConfig]);
```

Comment:
I guess you get an error in the console since this dependency array is missing `language`?
```suggestion
  }, [defaultConfig, language]);
```

Reply:
That being said, I think the best solution is rather to not depend on `language` and do this: https://github.com/ServiceNow/azimuth/pull/428/files#r1103166073